components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/TemplateManagerImpl.java

```diff
@@ -238,4 +244,149 @@ private void validatePaginationParameters(Integer limit, Integer offset) throws
+        } catch (IOException e) {
+            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_RETRIEVE_TEMPLATE_BY_ID, e,
+                    templateId, getTenantDomainFromCarbonContext());        }
+    }
+
+    @Override
+    public List<Template> listTemplates(String templateType, Integer limit, Integer offset) throws
+            TemplateManagementException {
+
+        if (!isValidTemplateType(templateType)) {
+            throw handleClientException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_TEMPLATE_TYPE,
+                    templateType);
+        }
+        if (limit != null || offset != null) {
+            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_PAGINATION_NOT_SUPPORTED, null);
+        }
+        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
+        try {
+            Resources resourcesList = configManager.getResourcesByType(templateType);
+            return resourcesList.getResources().stream().map(resource -> {
+                resource.setResourceType(templateType);
+                return new ResourceToTemplate().apply(resource);
+            }).collect(Collectors.toList());
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_LIST_TEMPLATES, e, templateType,
+                    getTenantDomainFromCarbonContext());
+        }
+    }
+
+    @Override
+    public void deleteTemplateById(String templateId) throws TemplateManagementException {
+
+        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
+        try {
+            configManager.deleteResourceById(templateId);
+        } catch (ConfigurationManagementException e) {
+            if (ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_ID_DOES_NOT_EXISTS.getCode().equals(
+                    e.getErrorCode())) {
+                throw handleClientException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_TEMPLATE_NOT_FOUND, e,
+                        templateId, getTenantDomainFromCarbonContext());
+            }
+            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_DELETE_TEMPLATE_BY_ID, e,
+                    templateId, getTenantDomainFromCarbonContext());
+        }
+    }
+
+    private String addTemplateToConfigStore(Template template) throws TemplateManagementException {
+
+        if (!isValidTemplateType(template.getTemplateType().toString())) {
+            throw handleClientException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_TEMPLATE_TYPE,
+                    template.getTemplateType().toString());
+        }
+        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
+        try {
+            Resource resource = configManager.addResource(template.getTemplateType().toString(), new
```

Comment:
How does the `template.getTemplateType()` is created?

Reply:
There's an attribute in the Template object to store the template type. This can take a one of the enums that are defined. During the execution of DB scripts, we need to add those types to the resource-type table.

Reply:
We may need to implement to make sure that the corresponding resource type always do exist in the DB. Ex: When creating the template, if the resource type does not exist, it can be created on the fly.

Reply:
Added with commit https://github.com/wso2/carbon-identity-framework/pull/2827/commits/105acf9bfe2a8d8ba258c0986ea7f1596cccee94